src/components/date_picker/super_date_picker/quick_select_popover/quick_select_popover.tsx

```diff
@@ -92,31 +97,59 @@ export class EuiQuickSelectPopover extends Component<
       recentlyUsedRanges,
       start,
       timeOptions,
+      customQuickSelectRender,
     } = this.props;
     const { prevQuickSelect } = this.state;
 
-    return (
-      <Fragment>
-        <EuiQuickSelect
-          applyTime={this.applyTime}
-          start={start}
-          end={end}
-          prevQuickSelect={prevQuickSelect}
-          timeOptions={timeOptions}
-        />
+    const commonlyUsedRangesElement = (
+      <>
         {commonlyUsedRanges.length > 0 && <EuiHorizontalRule margin="s" />}
         <EuiCommonlyUsedTimeRanges
           applyTime={this.applyTime}
           commonlyUsedRanges={commonlyUsedRanges}
         />
+      </>
+    );
+
+    const recentlyUsedRangesElement = (
+      <>
         {recentlyUsedRanges.length > 0 && <EuiHorizontalRule margin="s" />}
         <EuiRecentlyUsed
           applyTime={this.applyTime}
           commonlyUsedRanges={commonlyUsedRanges}
           dateFormat={dateFormat}
           recentlyUsedRanges={recentlyUsedRanges}
         />
-        {this.renderCustomQuickSelectPanels()}
+      </>
+    );
+
+    const customQuickSelectPanelsElement = (
+      <>{this.renderCustomQuickSelectPanels()}</>
+    );
+
+    return (
+      <Fragment>
+        <EuiQuickSelect
+          applyTime={this.applyTime}
+          start={start}
+          end={end}
+          prevQuickSelect={prevQuickSelect}
+          timeOptions={timeOptions}
+        />
+
+        {customQuickSelectRender ? (
+          customQuickSelectRender(
+            commonlyUsedRangesElement,
+            recentlyUsedRangesElement,
+            customQuickSelectPanelsElement
+          )
```

Comment:
API/naming suggestion - I like being specific with our own internal names, but I think what we pass to the the consumer can be shorter + should be in an obj

```suggestion
          customQuickSelectRender({
            quickSelect: quickSelectElement,
            commonlyUsed: commonlyUsedRangesElement,
            recentlyUsed: recentlyUsedRangesElement,
            customPanels: customQuickSelectPanelsElement,
          })
```